src/nss-myhostname/nss-myhostname.c

```diff
@@ -362,6 +369,8 @@ enum nss_status _nss_myhostname_gethostbyname3_r(
                 local_address_ipv4 = LOCALADDRESS_IPV4;
         }
 
+        UNPROTECT_ERRNO;
+
         return fill_in_hostent(
```

Comment:
So the rule is to call `UNPROTECT_ERRNO` before setting `*errnop`. `fill_in_hostent` sets `*errnop`, so it needs the same treatment. Seems reasonable.